var/spack/repos/builtin/packages/hipsycl/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+from os import path
+from llnl.util import filesystem
+
+
+class Hipsycl(CMakePackage):
+    """hipSYCL is an implementation of the SYCL standard programming model
+    over NVIDIA CUDA/AMD HIP"""
+
+    homepage = "https://github.com/illuhad/hipSYCL"
+    url = "https://github.com/illuhad/hipSYCL/archive/v0.8.0.tar.gz"
+    git = "https://github.com/illuhad/hipSYCL.git"
+
+    maintainers = ["nazavode"]
+
+    provides("sycl")
+
+    version("master", branch="master")
+    version(
+        "0.8.0",
+        sha256="4ec5218103d1b38efae9a51ca752b9b44bbd02dada78c05e20e00c9c25e9ea19",
+    )
+
+    variant("cuda", default=False, description="Enable CUDA backend for SYCL kernels")
+
+    depends_on("boost +filesystem")
+    depends_on("python@3:")
+    depends_on("llvm@8: +clang", when="~cuda")
+    # LLVM debug builds don't work with hipSYCL CUDA backend:
+    # https://github.com/illuhad/hipSYCL/blob/master/doc/install-cuda.md
+    depends_on("llvm@9: +clang build_type=Release", when="+cuda")
```

Comment:
@alalazo is there a way to say *any `build_type` except for `Debug`*?

Reply:
I'm not sure if there's a way to do it, but what would be even better would be to check if llvm was compiled with `-DLLVM_ENABLE_ASSERTIONS=OFF`, since this is the actual cmake flag that matters.

Reply:
> is there a way to say any build_type except for Debug?

You can use a conflict:
```python
conflicts('^llvm build_type=Debug', msg='explanation why')
```
Right now the line above will emit an error message if there's `llvm build_type=Debug` in the DAG. Once the concretizer is improved the information in `conflicts` should be used to exclude parts of the search space.

Reply:
> check if llvm was compiled with -DLLVM_ENABLE_ASSERTIONS=OFF

Current `llvm` package doesn't expose such fine grained controls, but since users aren't able to fiddle directly with `cmake` definitions I think we can consider `build_type != Debug` as a reasonable proxy.

Reply:
@alalazo added the `conflicts` directive to avoid `Debug` builds, hope the current concretizer will be happy.

Reply:
@nazavode Currently the concretizer disregards conflicts during constraint resolution and uses them after concretization to validate the spec. If the `spec` is not valid Spack displays an error with your custom message. At that point the user needs to specify better the abstract request to avoid the conflict.